packages/material-ui/src/Button/Button.js

```diff
@@ -82,29 +82,27 @@ export const styles = theme => ({
     border: `1px solid ${
       theme.palette.type === 'light' ? 'rgba(0, 0, 0, 0.23)' : 'rgba(255, 255, 255, 0.23)'
     }`,
+    '&$disabled': {
+      border: `1px solid ${theme.palette.action.disabled}`,
+    },
   },
   /* Styles applied to the root element if `variant="outlined"` and `color="primary"`. */
   outlinedPrimary: {
     color: theme.palette.primary.main,
-    border: `1px solid ${fade(theme.palette.primary.main, 0.5)}`,
+    border: '1px solid currentColor',
```

Comment:
The border is intended to be semi-transparent, so this change introduces a visual regression.

Reply:
I have the same concerns

Reply:
The advantage is: one less color variant = one less color to override. How would you solve the problem?

Reply:
If I revert the changes to primary and secondary (except the disabled change), inherit works fine, with the unavoidable limitation that the border opacity is not controlled. So the fix is simply `borderColor: 'currentColor'` in `colorInherit` as you already have it.

Reply:
👍